x-pack/plugins/asset_manager/server/routes/utils.ts

```diff
@@ -16,3 +21,26 @@ export async function getClientsFromContext<T extends RequestHandlerContext>(con
     savedObjectsClient: coreContext.savedObjects.client,
   };
 }
+
+type ValidateStringAssetFiltersReturn =
+  | [{ error: RouteValidationError }]
+  | [null, AssetFilters | undefined];
+
+export function validateStringAssetFilters(
+  q: any,
+  res: RouteValidationResultFactory
+): ValidateStringAssetFiltersReturn {
+  if (typeof q.stringFilters === 'string') {
+    try {
+      const parsedFilters = JSON.parse(q.stringFilters);
+      if (assetFiltersSingleKindRT.is(parsedFilters)) {
```

Comment:
I thought this would return false if any properties of the parseFilters don't exist on the type but it still returns true. For example `GET /api/asset-manager/assets/hosts?from=now-1m&to=now&stringFilters=%7B%22foobar%22%3A%22baz%22%7D` (`filters={"foobar": "baz"}`) passes through and returns all the services, is that expected ?